src/signTx.h

```diff
@@ -166,9 +166,15 @@ ins_sign_tx_aux_data_context_t* accessAuxDataContext();
 ins_sign_tx_body_context_t* accessBodyContext();
 ins_sign_tx_witness_context_t* accessWitnessContext();
 
-#define AUX_DATA_CTX() (ASSERT(accessAuxDataContext() != NULL), accessAuxDataContext())
-#define BODY_CTX() (ASSERT(accessBodyContext() != NULL), accessBodyContext())
-#define WITNESS_CTX() (ASSERT(accessWitnessContext() != NULL), accessWitnessContext())
+#ifdef DEVEL
+#define DEBUG_ONLY_ASSERT(cond) ASSERT(cond),
+#else
+#define DEBUG_ONLY_ASSERT(cond) 
+#endif
+
+#define AUX_DATA_CTX() (DEBUG_ONLY_ASSERT(accessAuxDataContext() != NULL) accessAuxDataContext())
+#define BODY_CTX() (DEBUG_ONLY_ASSERT(accessBodyContext() != NULL) accessBodyContext())
+#define WITNESS_CTX() (DEBUG_ONLY_ASSERT(accessWitnessContext() != NULL) accessWitnessContext())
```

Comment:
I'd try to avoid "incomplete" macros (ending with an operator), e.g. like this:
```
#ifdef DEVEL
#define ASSERT_DEVEL_ONLY(cond) ASSERT(cond)
#else
#define ASSERT_DEVEL_ONLY(cond) 0
#endif

#define AUX_DATA_CTX() (ASSERT_DEVEL_ONLY(accessAuxDataContext() != NULL), accessAuxDataContext())
```
Otherwise, I like it, it seems to solve all our problems (no need to rename the access* functions).

Reply:
But maybe doing it directly is even clearer?
```
#ifdef DEVEL
#define AUX_DATA_CTX (ASSERT(accessAuxDataContext() != NULL), accessAuxDataContext())
#else
#define AUX_DATA_CTX (accessAuxDataContext())
#endif

```

Reply:
The first version was the one I tried, it produces a ton of warnings about unused values (the 0), so I ditched that, but yeah, your second version is actually the good solution I think, not sure why I didn't think of that.